### PR TITLE
Document indexing behavior of nvim_win_get_cursor

### PR DESCRIPTION
**Pull request by KillTheMule**
I stumbled over this, so I figured I'd document it.

@justinmk suggested a table of indexing behavior in api.txt. I'd do it, but let me point out the functions that would be mentioned:

- Functions where lines/columns are 0-indexed, and ranges are end-exclusive:
  - nvim_buf_lines_event[{buf}, {changedtick}, **{firstline}, {lastline}**, {linedata}, {more}]
  - nvim_buf_get_lines({buffer}, **{start}, {end}**, {strict_indexing})
  - nvim_buf_set_lines({buffer}, **{start}, {end}**, {strict_indexing},
  - nvim_buf_add_highlight({buffer}, {ns_id}, {hl_group}, **{line},{col_start}, {col_end}**)
  - nvim_buf_clear_namespace({buffer}, {ns_id}, **{line_start}, {line_end}**)
  - nvim_buf_get_offset({buffer}, **{index}**)
- Functions that exhibit "mark-like" behavior (lines 1-indexed, columns 0-indexed)
  - nvim_buf_get_mark({buffer}, {name})                     
  - nvim_win_get_cursor({window})                     
  - nvim_win_set_cursor({window}, {pos})                  

Should I put that into a table? Most of it is documented in the function docs themselves (I'd recheck if we don't do the table), and there's just 2 types of behavior, "api-like" and "mark-like". I'd suggest we just add a small paragraph with a sentence, rather than a table.

### CODE REVIEW

**Comment by justinmk** [comment]
LGTM, let's try the table also.  Basically just copy-paste what you wrote in this issue, but list tags instead of the signatures:

    |nvim_buf_lines_event|
    |nvim_buf_get_lines()|

We can always remove it later if it feels too verbose. This section can go in the `:help api-types` section, we can rename that section to something more generic like `api-definitions`.

File: runtime/doc/api.txt [hunk 1]
```diff
@@ -34,6 +36,27 @@ discriminated as separate types in an Object:
...
+
+    |nvim_buf_lines_event|
+    |nvim_buf_get_lines()|
+    |nvim_buf_set_lines(),|
+    |nvim_buf_add_highlight()|
+    |nvim_buf_clear_namespace()|
+    |nvim_buf_get_offset()|
+
+For historical reasons, the following functions inhibit "mark-like" behavior,
+i.e. lines are 1-indexed, columns are 0-indexed.
```
Comment (justinmk) [inline runtime/doc/api.txt]: s/inhibit/exhibit ?

Comment (KillTheMule) [reply runtime/doc/api.txt]: Thanks :)

**Comment by justinmk** [comment]
Merged, thanks @KillTheMule !

> I'd suggest we just add a small paragraph with a sentence, rather than a table.

I agree it doesn't make sense to list every function that follows the "normal rules". So I only left the "exceptions" table.

**Pull request closed without merging**